docs/DeveloperGuide.md

```diff
@@ -1,33 +1,230 @@
+- [Product Scope](#product-scope)
+    - [Target User Profile](#target-user-profile)
+    - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-functional-requirements)
+- [Manual Testing](#instructions-for-manual-testing)
+- [Command Summary](#command-summary)
+- [Glossary](#glossary)
+- [Acknowledgements](#acknowledgements)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+## Introduction
+**ActiveEdge** is a program for a university students to manage their food intake, water intake and exercises via the Command Line Interface (CLI).
+### Purpose
+This document specifies the architectural and software design decisions in the implementation of **ActiveEdge**.
+
+
+### How to Use a Developer Guide
+
+Are you a newcomer?
+No problem, navigate to the Quick Start section.
+Feeling disoriented within the content?
+Refer to the Table of Contents to find the appropriate pages.
+Require assistance with the features?
+Consult the Features section in the User Guide for thorough guidance.
+Have inquiries for us?
+Visit the FAQ section in the User Guide.
+Seeking a brief overview of all features?
+Check out the Command Summary page for a summary of all commands.
+Uncertain about a specific term's meaning?
+Refer to the Glossary page for clarification.
+
+## Quick Start
+
+To get started quickly with ActiveEdge,
+follow these simple steps:
+
+1. Ensure that you have Java 11 or above installed.
+2. Down the latest version of `ActiveEdge` from [here](http://link.to/duke). 
+3. Copy the file into the folder you want to use as the home folder for your ActiveEdge.
+4. Open a command terminal, cd into the home folder you put the jar file in, and run the following command: java -jar ActiveEdge.jar. You should see a welcome screen.
+5. Type commands beside >>> and press Enter to execute it.
+(e.g. typing ```help``` and pressing Enter will show the help page).
+## Design & implementation 🏹  
+### System Architecture
+
+![System Architecture](../images/System-Architecture.png)
+
+The Architecture Diagram above shows a high-level overview of the architectural design of ActiveEdge.
+
+Active Edge is comprised of 8 major components:
+
+```Main```: The entry point of the application.\
+```UI```: A package responsible for handling user interactions, both capturing input and displaying output.\
+```Parser```: A package that analyzes and interprets the user's input commands.\
+```Command```: A package tasked with executing specific actions based on the user's commands.\
+```Storage```: A package focused on persisting and retrieving all relevant data from the application's local storage system, ensuring data continuity and integrity.\
+```LogList```: Maintains logged data throughout the application's runtime.\
+```FoodData```: Active Edge's food database\
+```ExerciseData```: Active Edge's exercise database\
```

Comment:
ExerciseData line has a slash at the end appears in the DG. Maybe remove?

Reply:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111181615/b94e3080-a906-4cf9-92cd-f498fb438be8)